ckanext/googleanalytics/commands.py

```diff
@@ -109,129 +113,205 @@ def load_analytics(self, args):
-        from ga_auth import (init_service, get_profile_id)
+        from ga_auth import get_profile_id
 
-        if len(args) == 1:
-            raise Exception("Missing token file")
-        tokenfile = args[1]
-        if not os.path.exists(tokenfile):
-            raise Exception('Cannot find the token file %s' % args[1])
+        self.init_service(args)
 
-        try:
-            self.service = init_service(args[1], None)
-        except TypeError:
-            print ('Have you correctly run the getauthtoken task and '
-                   'specified the correct file here')
-            raise Exception('Unable to create a service')
         self.profile_id = get_profile_id(self.service)
         if len(args) > 3:
             raise Exception('Too many arguments')
 
         given_start_date = None
         if len(args) == 3:
             given_start_date = datetime.datetime.strptime(args[2], '%Y-%m-%d').date()
-      
-        packages_data = self.get_ga_data(start_date=given_start_date)
-        self.save_ga_data(packages_data)
-        self.log.info("Saved %s records from google" % len(packages_data))
 
-    def save_ga_data(self, packages_data):
+        # list of queries to send to analytics
+        queries = [{
+            'type': 'package',
+            'dates': self.get_dates_between_update(given_start_date, PackageStats.get_latest_update_date()),
+            'filters': 'ga:pagePath=~%s,ga:pagePath=~%s' % (PACKAGE_URL, self.resource_url_tag),
+            'metrics': 'ga:uniquePageviews',
+            'sort': '-ga:uniquePageviews',
+            'dimensions': 'ga:pagePath, ga:date',
+        }, {
+            'type': 'visitorlocation',
+            'dates': self.get_dates_between_update(given_start_date, AudienceLocationDate.get_latest_update_date()),
+            'filters': None,
+            'metrics': 'ga:sessions',
+            'sort': '-ga:sessions',
+            'dimensions': 'ga:country, ga:date',
+        }]
+
+        queries = [queries[1]]
+
+        # loop through queries, parse and save them to db
+        for query in queries:
+            data = {}
+            current = datetime.datetime.now()
+            print 'performing analytics query of type: %s' % query['type']
+            for date in query['dates']:
+                # run query with current query values
+                results = self.ga_query(start_date=date,
+                                        end_date=current,
+                                        filters=query['filters'],
+                                        metrics=query['metrics'],
+                                        sort=query['sort'],
+                                        dimensions=query['dimensions'])
+                
+                # parse query
+                data = self.query_resolvers(query['type'], results, data)
+                current = date
+            
+            self.save_ga_data(query['type'], data)
+
+    def save_ga_data(self, querytype, data):
```

Comment:
This function seems to have two distinct modes that have little to do with each other. Maybe split stuff inside top level if into functions for readability?

Reply:
Also, shouldn't the latter be elif?